src/main/kotlin/gg/skytils/skytilsmod/utils/MojangUtil.kt

```diff
@@ -65,6 +73,13 @@ object MojangUtil {
         }
     }
 
+    suspend fun queueUsernameForUUID(name: String): UUID? {
+        val username = name.lowercase()
+        usernameQueue.emit(username to System.currentTimeMillis())
+
+        return uuidFlow.first { it.first == username }.second
+    }
```

Comment:
Is there any guarantee that the username will be resolved? Afaict you need to pray someone else calls `MojangUtil#getUsernameFromUUID`.